js/src/edit-block/hooks/useField.js

```diff
@@ -275,6 +276,8 @@ const useField = () => {
 		}
 
 		if ( newSettings.hasOwnProperty( 'name' ) ) {
+			// If the new name already exists like 'foo', append a space like 'foo '
+			newSettings.name = newBlock.fields[ newSettings.name ] ? `${ newSettings.name } ` : newSettings.name;
```

Comment:
This is a little hacky. But it's much simpler than [previous](https://github.com/studiopress/genesis-custom-blocks/pull/56/commits/c6e7cc62a2205f65fe884d99fb0c19ba82ecda9c) [attempts](https://github.com/studiopress/genesis-custom-blocks/pull/56/commits/3cab15fb343b681a388f578f077218da820f6d07).

The problem is that no 2 fields can have the same name.

This is because fields are keyed by name in the `fields` array.

Like:

```js
{
        app: {location: "editor", width: "100", help: "", default: "", placeholder: "", …}
       "app name": {location: "editor", width: "100", help: "", default: "", placeholder: "", …}
}
```

So those keys couldn't both be "app".